src/library/scala/util/matching/Regex.scala

```diff
@@ -768,10 +768,30 @@ object Regex {
    */
   object Groups {
     def unapplySeq(m: Match): Option[Seq[String]] = {
-      if (m.groupCount > 0) Some(List.tabulate(m.groupCount)(i => m.group(i + 1))) else None
+      if (m.groupCount > 0) extractGroupsFromMatch(m) else None
     }
   }
 
+  def extractGroupsFromMatch(m: Match): Option[List[String]] = {
+    var res = List.empty[String]
+    var index = m.groupCount
+    while (index >0 ) {
+      res ::= m.group(index)
+      index -= 1
+    }
+    Some(res)
+  }
+
+  def extractGroupsFromMatcher(m: Matcher): Option[List[String]] = {
```

Comment:
Let's make these new methods private

Reply:
yes I find this break the migration manager check.